tests/e2e_tests/spec_l3_to_l4_psfsub_e2e.py

```diff
@@ -0,0 +1,304 @@
+    l3_target = sorted(glob.glob(os.path.join(target_l3_output_dir, "cgi_*l3_.fits")))
+    l3_files.extend(l3_target)
+    l3_dataset = Dataset(l3_files)
+        
+    logger.info('')
+    
+    # ================================================================================
+    # (2) Validate Input Images
+    # ================================================================================
+    logger.info('='*80)
+    logger.info('Test Case 1: Input Image Data Format and Content')
+    logger.info('='*80)
+
+    # Validate all input images
+    for i, frame in enumerate(l3_dataset):
+        frame_info = f"L3 Frame {i}"
+        
+        check_filename_convention(getattr(frame, 'filename', None), 'cgi_*_l3_.fits', frame_info, logger, data_level = 'l3_')
+        check_dimensions(frame.data, (125, 125), frame_info, logger)
+        verify_header_keywords(frame.ext_hdr, {'DPAMNAME', 'CFAMNAME', 'FSAMNAME'}, frame_info, logger)
+        verify_header_keywords(frame.ext_hdr, {'DATALVL': 'L3'}, frame_info, logger)
+        verify_header_keywords(frame.pri_hdr, {'PSFREF'}, frame_info, logger)
+        logger.info("")
+
+    logger.info(f"Total input images validated: {len(l3_dataset)}")
+    logger.info("")
+    
+    # Create a temporary caldb and add the default DispersionModel calibration
+    tmp_caldb_csv = os.path.join(corgidrp.config_folder, 'tmp_e2e_test_caldb.csv')
+    corgidrp.caldb_filepath = tmp_caldb_csv
+    # remove any existing caldb file so that CalDB() creates a new one
+    if os.path.exists(corgidrp.caldb_filepath):
+        os.remove(tmp_caldb_csv)
+    this_caldb = caldb.CalDB()
+    
+    # Create calibrations subfolder
+    calibrations_dir = os.path.join(e2eoutput_path, 'calibrations')
+    if not os.path.exists(calibrations_dir):
+        os.makedirs(calibrations_dir)
+    #Create a mock flux calibration file
+    fluxcal_factor = 2e-12
+    fluxcal_factor_error = 1e-14
+    prhd, exthd, errhd, dqhd = create_default_calibration_product_headers()
+    # Set consistent header values for flux calibration factor
+    exthd['CFAMNAME'] = '3'
+    exthd['DPAMNAME'] = 'PRISM3'
+    exthd['FSAMNAME'] = 'R1C2'
+    fluxcal_fac = corgidrp.data.FluxcalFactor(fluxcal_factor, err = fluxcal_factor_error, pri_hdr = prhd, ext_hdr = exthd, err_hdr = errhd, input_dataset = l3_dataset)
+
+    rename_files_to_cgi_format(list_of_fits=[fluxcal_fac], output_dir=calibrations_dir, level_suffix="abf_cal")
+    this_caldb.create_entry(fluxcal_fac)
+    
+    # Scan for default calibrations
+    this_caldb.scan_dir_for_new_entries(corgidrp.default_cal_dir)
+    
+    # ================================================================================
+    # (3) Run Processing Pipeline
+    # ================================================================================
+    logger.info('='*80)
+    logger.info('Running processing pipeline')
+    logger.info('='*80)
+
+    logger.info('Running e2e recipe...')
+    recipe = walk_corgidrp(
```

Comment:
just like for the other e2e test, we should modify the walker to guess the recipe, so we don't need to manually specify the template here. Again, can use the logic already used in walker.guess_recipe to identify spec and coronagraphic data.